### PR TITLE
Enforce print convention: No leading "A" in printouts

### PR DESCRIPTION
**Pull request by HereAround**
As discussed in the OSCAR meeting last Friday, printouts should not begin with "A" (or "An"). I have looked through the `src` folder of OSCAR and found that this happened for objects created by:
* toric geometry,
* tropical geometry,
* polyhedral geometry.

This PR removes the "A" in the printout messages from these areas and adjusts the doctests accordingly. Notice that the doctests were corrected via `OSCAR.build_doc(doctest=:fix)`. So most of the code in this PR was automatically generated. To me, it looked alright.

cc @fingolfin @fieker 

~~(For my convenience, this includes the changes in https://github.com/oscar-system/Oscar.jl/pull/1970, so that possibly no non-trivial `rebase` is needed.)~~ (This PR was just merged.)

### CODE REVIEW

**Comment by HereAround** [comment]
@fingolfin and @fieker : This should be ready for review.

**Comment by HereAround** [comment]
@afkafkafk13 You might like this.

File: src/PolyhedralGeometry/Polyhedron/properties.jl [hunk 1]
```diff
@@ -1250,9 +1250,9 @@ print_constraints(H::SubObjectIterator{<:Hyperplane}; numbered::Bool = false, io
 function Base.show(io::IO, H::Halfspace)
     n = length(normal_vector(H))
     if iszero(normal_vector(H)) && negbias(H) >= 0
-        print(io, "The trivial Halfspace, R^$n")
```
Comment (afkafkafk13) [inline src/PolyhedralGeometry/Polyhedron/properties.jl:1253]: Here the definite article makes sense.

File: src/PolyhedralGeometry/Polyhedron/properties.jl [hunk 2]
```diff
@@ -1250,9 +1250,9 @@ print_constraints(H::SubObjectIterator{<:Hyperplane}; numbered::Bool = false, io
...
+        print(io, "Trivial Halfspace, R^$n")
     else
-        print(io, "The Halfspace of R^$n described by\n")
```
Comment (afkafkafk13) [inline src/PolyhedralGeometry/Polyhedron/properties.jl:1255]: Here the definitie article makes sense.
(and two more occurrences in this file)

File: src/PolyhedralGeometry/Polyhedron/properties.jl [hunk 3]
```diff
@@ -1261,9 +1261,9 @@ function Base.show(io::IO, H::Hyperplane)
     n = length(normal_vector(H))
     b = negbias(H)
     if iszero(b) && iszero(normal_vector(H))
-        print(io, "The trivial Hyperplane, R^$n")
+        print(io, "Trivial Hyperplane, R^$n")
```
Comment (fingolfin) [inline src/PolyhedralGeometry/Polyhedron/properties.jl]: Please change "Hyperplane" to "hyperplane", and "Halfspace" to "half-space"

File: src/TropicalGeometry/hypersurface.jl [hunk 4]
```diff
@@ -165,12 +165,12 @@ julia> Kx, (x1,x2) = PolynomialRing(QQ,2)
 (Multivariate Polynomial Ring in x1, x2 over Rational Field, fmpq_mpoly[x1, x2])
 
 julia> val = TropicalSemiringMap(QQ,7)
-The 7-adic valuation on Rational Field
```
Comment (afkafkafk13) [inline src/TropicalGeometry/hypersurface.jl]: This again should stay a 'the'.

File: src/TropicalGeometry/poly.jl [hunk 5]
```diff
@@ -68,7 +68,7 @@ julia> R, (x,y) = PolynomialRing(QQ,["x", "y"])
 (Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
 
 julia> val = TropicalSemiringMap(QQ,7)
-The 7-adic valuation on Rational Field
```
Comment (afkafkafk13) [inline src/TropicalGeometry/poly.jl]: see above

File: src/TropicalGeometry/linear_space.jl [hunk 6]
```diff
@@ -47,10 +47,10 @@ julia> I = ideal(R,[-x1+x3+x4,-x2+x3+x5,-x1+x2+x6])
 ideal(-x1 + x3 + x4, -x2 + x3 + x5, -x1 + x2 + x6)
 
 julia> val = TropicalSemiringMap(ZZ)
-The trivial valuation on Integer Ring
```
Comment (afkafkafk13) [inline src/TropicalGeometry/linear_space.jl:50]: 'the' is fine here.

File: src/TropicalGeometry/valuation.jl [hunk 7]
```diff
@@ -118,7 +118,7 @@ end
 
 # Display:
 function Base.show(io::IO, val::TropicalSemiringMap{K,Nothing} where {K})
-    print(io, "The trivial valuation on $(val.valued_field)")
```
Comment (afkafkafk13) [inline src/TropicalGeometry/valuation.jl]: see above

File: src/TropicalGeometry/valuation.jl [hunk 8]
```diff
@@ -150,7 +150,7 @@ end
 
 # Display:
 function Base.show(io::IO, val::TropicalSemiringMap{FlintRationalField,fmpq})
-    print(io, "The $(val.uniformizer_field)-adic valuation on $(val.valued_field)")
```
Comment (afkafkafk13) [inline src/TropicalGeometry/valuation.jl]: again 'the' ?

**Review by afkafkafk13 — COMMENTED** [review]
Thank you for all the work you are putting into this. 

Please have a look at the instances of definite articles which I marked. There we do not have the context dependence of the need for 'a' or 'the', which we avoid by dropping the article, and hence we can leave (the mathematically correct) 'the'.

File: src/PolyhedralGeometry/Polyhedron/properties.jl (hunk 1 above)
Comment (HereAround) [reply src/PolyhedralGeometry/Polyhedron/properties.jl:1253]: Changed.

File: src/PolyhedralGeometry/Polyhedron/properties.jl (hunk 2 above)
Comment (HereAround) [reply src/PolyhedralGeometry/Polyhedron/properties.jl:1255]: Changed.

File: src/PolyhedralGeometry/Polyhedron/properties.jl (hunk 3 above)
Comment (HereAround) [reply src/PolyhedralGeometry/Polyhedron/properties.jl]: Done.

File: src/TropicalGeometry/hypersurface.jl (hunk 4 above)
Comment (HereAround) [reply src/TropicalGeometry/hypersurface.jl]: Ok. Done.

File: src/TropicalGeometry/poly.jl (hunk 5 above)
Comment (HereAround) [reply src/TropicalGeometry/poly.jl]: Also adjusted.

File: src/TropicalGeometry/linear_space.jl (hunk 6 above)
Comment (HereAround) [reply src/TropicalGeometry/linear_space.jl:50]: Done.

File: src/TropicalGeometry/valuation.jl (hunk 7 above)
Comment (HereAround) [reply src/TropicalGeometry/valuation.jl]: Done.

File: src/TropicalGeometry/valuation.jl (hunk 8 above)
Comment (HereAround) [reply src/TropicalGeometry/valuation.jl]: Ok. Done.

**Comment by HereAround** [comment]
@afkafkafk13 I have just addressed the changes that you requested, i.e. have installed "The" in the places you marked. I hope that the tests will pass. But since they take a while, please feel free to inspect again. Thank you!

**Comment by afkafkafk13** [comment]
Looks good now.
If @fingolfin does not have further remarks (from the broader perspective), this should be good to go.

**Review by afkafkafk13 — APPROVED** [review]

**Comment by HereAround** [comment]
I have mentioned this change in the OSCAR meeting today (no complaints), @fingolfin busy with the renaming and all tests pass here. So I think this is safe to merge (and so I can rebase https://github.com/oscar-system/Oscar.jl/pull/1977).

**Pull request merged**